File: search_148.py
# Search for '148' in 'app.js'
import os

filepath = r'c:\Users\Migue\Desktop\persiana-total-app\app.js'
if os.path.exists(filepath):
    with open(filepath, 'r', encoding='utf-8', errors='replace') as f:
        for i, line in enumerate(f, 1):
            if '148' in line:
                print(f"{i}: {line.strip()}")
else:
    print(f"File {filepath} not found.")

File: search_148.js
const fs = require('fs');
const readline = require('readline');

async function search() {
    const fileStream = fs.createReadStream('c:\\Users\\Migue\\Desktop\\persiana-total-app\\app.js');
    const logsStream = fs.createWriteStream('c:\\Users\\Migue\\Desktop\\persiana-total-app\\search_results.txt');
    const rl = readline.createInterface({
        input: fileStream,
        crlfDelay: Infinity
    });

    let lineNumber = 0;
    for await (const line of rl) {
        lineNumber++;
        if (line.includes('148')) {
            logsStream.write(`${lineNumber}: ${line.trim()}\n`);
        }
    }
    logsStream.end();
}

search();
